### PR TITLE
TRACEFOSS-135 - Added new page for parts. Added table components

### PR DESCRIPTION
**Pull request by evia-fbedarf**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -30,9 +30,6 @@ RUN npm run ng build -- --configuration=${PROFILE} --output-path=dist
 FROM nginxinc/nginx-unprivileged:alpine
 # Remove NGINX default configuration
 RUN rm /etc/nginx/conf.d/default.conf
-# Ensure that /tmp exists and user has access to it
-
-VOLUME ["/tmp"]
```
Comment (ds-ext-akononenko) [inline Dockerfile:35]: can you please revert this change from this file?

File: src/app/modules/page/parts/core/parts.assembler.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Part, PartResponse } from '@page/parts/model/parts.model';
+
+export class PartsAssembler {
+  public static assembleParts(parts: PartResponse[]): Part[] {
+    const transformedParts: Part[] = [];
+    parts.forEach(part => {
```
Comment (ds-ext-akononenko) [inline src/app/modules/page/parts/core/parts.assembler.ts:25]: can you please change this to `.map`?

File: src/app/modules/page/parts/model/parts.model.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  manufacturerName: string;
+  nameAtCustomer: string;
+  customerPartId: string;
+  manufacturingDate: string;
+  manufacturingCountry: string;
+  specificAssetIds: Record<string, string>;
+  childDescriptions: Array<{ id: string; idShort: string }>;
+}
+
+export interface PartsResponse extends Array<PartResponse> {}
```
Comment (ds-ext-akononenko) [inline src/app/modules/page/parts/model/parts.model.ts:49]: maybe just `type PartsResponse = PartResponse[]`?

File: src/app/modules/page/parts/core/parts.assembler.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      const transformedPart = {} as Part;
+      transformedPart.id = part.id;
+      transformedPart.name = part.nameAtManufacturer;
+      transformedPart.manufacturer = part.manufacturerName;
+      transformedPart.serialNumber = part.manufacturerPartId;
+      transformedPart.partNumber = part.customerPartId;
+      transformedPart.productionCountry = part.manufacturingCountry;
+      transformedPart.qualityType = 'high';
+      transformedPart.productionDate = part.manufacturingDate;
```
Comment (ds-ext-akononenko) [inline src/app/modules/page/parts/core/parts.assembler.ts:34]: can you please map here `productionDate` to actual Date?

File: src/app/modules/page/parts/model/parts.model.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+export interface Part {
+  id: string;
+  name: string;
+  manufacturer: string;
+  serialNumber: string;
+  partNumber: string;
+  productionCountry: string;
+  qualityType: QualityType;
+  productionDate: string;
```
Comment (ds-ext-akononenko) [inline src/app/modules/page/parts/model/parts.model.ts:30]: I believe this should have `Date` type

File: src/app/modules/page/parts/presentation/parts.component.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    this.parts$ = this.partsFacade.parts$;
+    this.tableConfig = { displayedColumns: this.displayedColumns, sortableColumns: this.sortableColumns };
+  }
+
+  ngOnInit(): void {
+    this.partsFacade.setParts();
+  }
+
+  public onSelectItem($event: Record<string, unknown>) {
+    console.log($event);
```
Comment (ds-ext-akononenko) [inline src/app/modules/page/parts/presentation/parts.component.ts:72]: do we want to commit into main?

File: src/app/modules/page/parts/core/parts.assembler.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Part, PartResponse } from '@page/parts/model/parts.model';
+
+export class PartsAssembler {
+  public static assembleParts(parts: PartResponse[]): Part[] {
```
Comment (ds-ext-akononenko) [inline src/app/modules/page/parts/core/parts.assembler.ts:23]: do we want to create empty classes which has only static function?

I mean you can just define function on module level and export from here, or we can make this method non-static and use like normal method.

File: src/app/mocks/services/parts-mock/parts.model.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  {
+    id: '1',
+    idShort: '1',
+    nameAtManufacturer: 'Audi A1 Sportback',
+    nameAtCustomer: 'Audi A1 Sportback',
+    manufacturerId: '1',
+    manufacturerName: 'Audi',
+    customerPartId: '11111',
+    manufacturerPartId: '5XXGM4A77CG032209',
+    manufacturingDate: '1997-05-30T12:34:12',
```
Comment (ds-ext-akononenko) [inline src/app/mocks/services/parts-mock/parts.model.ts:31]: I hope backend would return datetime with timezone, or we would have a big problem

File: Dockerfile (hunk 1 above)
Comment (evia-fbedarf) [reply Dockerfile:35]: Don't even know why this was deleted

File: src/app/mocks/services/parts-mock/parts.model.ts (hunk 8 above)
Comment (evia-fbedarf) [reply src/app/mocks/services/parts-mock/parts.model.ts:31]: I discussed the data with Dirk, and we could see that the regex for this date does not include the timezone.
This is why I did not add it.

I will ask dirk for the documentation on that.

File: src/app/modules/page/parts/core/parts.assembler.ts (hunk 7 above)
Comment (evia-fbedarf) [reply src/app/modules/page/parts/core/parts.assembler.ts:23]: Yes, you could define a function on the module level, but it is all about the long term.

When the project is running for a year or two, you won't know where you started to manipulate the data.
You start to do it wherever it fits instead of keeping it all in the same structure.
That is why I would like to stick to the `assembler`, `facade`, `service`, `state` approach.

Assembler Manipulates data.
Facade Communicates between components and services.
Service Communicates with the API.
State stores the essential state for the current module.

File: src/app/modules/page/parts/core/parts.assembler.ts (hunk 2 above)
Comment (evia-fbedarf) [reply src/app/modules/page/parts/core/parts.assembler.ts:25]: Done.

File: src/app/mocks/services/parts-mock/parts.model.ts (hunk 8 above)
Comment (evia-fbedarf) [reply src/app/mocks/services/parts-mock/parts.model.ts:31]: This is the regex:
`"pattern": "-?([1-9][0-9]{3,}|0[0-9]{3})-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])T(([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\\.[0-9]+)?|(24:00:00(\\.0+)?))(Z|(\\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00))?"`

This is an example value:
'2022-02-04T14:48:54'

File: src/app/modules/page/parts/core/parts.assembler.ts (hunk 4 above)
Comment (evia-fbedarf) [reply src/app/modules/page/parts/core/parts.assembler.ts:34]: Done.

File: src/app/modules/page/parts/model/parts.model.ts (hunk 5 above)
Comment (evia-fbedarf) [reply src/app/modules/page/parts/model/parts.model.ts:30]: Done.

File: src/app/modules/page/parts/model/parts.model.ts (hunk 3 above)
Comment (evia-fbedarf) [reply src/app/modules/page/parts/model/parts.model.ts:49]: Done.

File: src/app/modules/page/parts/presentation/parts.component.ts (hunk 6 above)
Comment (evia-fbedarf) [reply src/app/modules/page/parts/presentation/parts.component.ts:72]: For now, this is fine. I left it in on purpose so we can test the API and data manipulation with the actual BE.
And so we know that a click is happening. This will be removed in the Relations Ticket.

File: src/app/mocks/services/parts-mock/parts.model.ts (hunk 8 above)
Comment (ds-ext-akononenko) [reply src/app/mocks/services/parts-mock/parts.model.ts:31]: ok, should we then assume that all date string in UTC?
